pkg/controllers/workgenerator/controller.go

```diff
@@ -530,7 +533,36 @@ func buildAllWorkAppliedCondition(works map[string]*fleetv1beta1.Work, binding *
 		Status:             metav1.ConditionFalse,
 		Type:               string(fleetv1beta1.ResourceBindingApplied),
 		Reason:             workNotAppliedReason,
-		Message:            "not all corresponding work objects are applied",
+		Message:            fmt.Sprintf("work object %s is not applied", notAppliedWork),
+		ObservedGeneration: binding.GetGeneration(),
+	}
+}
+
+func buildAllWorkAvailableCondition(works map[string]*fleetv1beta1.Work, binding *fleetv1beta1.ClusterResourceBinding) metav1.Condition {
```

Comment:
when crp populates the status, if the last condition is unknown or false, it won't populate the remaining in order to keep the our doc simple (easy to explain to user). Otherwise, we have too many combinations. 

Just wonder if we would like to keep them consistent?
so that we only populate the available condition when the applied condition is true.